### PR TITLE
Add OpenLDAP Tutorial

### PR DESCRIPTION
**Pull request by fhennig**
~~**blocked** by https://github.com/stackabletech/issues/issues/322 - The tutorial doesn't work because of the Trino and Superset inconsistencies~~ (It's still not working well, but we can still merge it)

belongs to #300 

The tutorial works with 22.11, not with nightly, because it uses Stacks. If the Stacks are updated, the tutorial should also still work, hopefully.

### CODE REVIEW

File: modules/tutorials/examples/ldap-auth/60-modify-trino.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+echo "Updating superset.yaml in-place with authentication LDAP snippet"
+yq -i 'del(.spec.authentication.method.multiUser)' trino.yaml
+yq -i '. *= load("trino-auth-snippet.yaml")' trino.yaml
+
+echo "Applying updated configuration"
+# tag::apply-cluster[]
+kubectl apply -f trino.yaml
+# end::apply-cluster[]
+
+echo "Waiting on Trino StatefulSets rollout ..."
```
Comment (adwk67) [inline modules/tutorials/examples/ldap-auth/60-modify-trino.sh:34]:
```suggestion
echo "Waiting for Trino StatefulSets rollout ..."
```

File: modules/tutorials/examples/ldap-auth/README.adoc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+= Example directory for OpenLDAP authentication tutorial
+
+`main.sh` executes all the other shell scripts in order (10-install-release.sh etc.).
+
+The numbered scripts correspond to steps in the tutorial, you can run them one by one
```
Comment (adwk67) [inline modules/tutorials/examples/ldap-auth/README.adoc:5]:
```suggestion
The numbered scripts correspond to steps in the tutorial: you can run them one by one 
```

File: modules/tutorials/examples/ldap-auth/20-verify-superset.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+set -euo pipefail
+shopt -s lastpipe
+
+source "./utils.sh"
+
+echo "Waiting on superset StatefulSet ..."
```
Comment (adwk67) [inline modules/tutorials/examples/ldap-auth/20-verify-superset.sh:7]:
```suggestion
echo "Waiting for superset StatefulSet ..."
```

File: modules/tutorials/pages/authentication_with_openldap.adoc [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+
+Prerequisites:
+
+* a k8s cluster
+* xref:stackablectl::index.adoc[] installed
+* basic knowledge of how to create resources in Kubernetes (i.e. `kubectl apply -f <filename>.yaml`) and inspect them (`kubectl get` or a tool like `k9s`)
+
+== Setup
+
+Before configuring LDAP you need to set up some services to configure, as well as an LDAP to use for authentication. This tutorial is about learning how to configure an LDAP, so there won't be detailed instructions on how to set up all of this, instead we will use xref:stackablectl::commands/stack.adoc[stackablectl Stacks].
```
Comment (adwk67) [inline modules/tutorials/pages/authentication_with_openldap.adoc:13]: Nit: we switch a bit between LDAP (the protocol) and **_an_** LDAP (installation): I would make that little more explicit e.g.
```suggestion
Before configuring LDAP you need to set up some services to configure, as well as an LDAP server to use for authentication. This tutorial is about learning how to configure LDAP, so there won't be detailed instructions on how to set up all of this, instead we will use xref:stackablectl::commands/stack.adoc[stackablectl Stacks]. 
```

File: modules/tutorials/pages/authentication_with_openldap.adoc [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+include::example$ldap-auth/10-install-base.sh[tag=stackablectl-install]
+
+This command will take a few minutes to complete.
+
+NOTE: The stack installed here is used in the xref:stackablectl::demos/trino-taxi-data.adoc[`trino-tax-data` demo]. Click the link to learn more.
+
+[#setup_login]
+==== Inspect
+
+Before starting with adding configuration to your Stackable cluster, inspect what the command above has set up in your Kubernetes cluster.
```
Comment (adwk67) [inline modules/tutorials/pages/authentication_with_openldap.adoc:29]:
```suggestion
Before starting to add configuration to your Stackable cluster, inspect what the command above has set up in your Kubernetes cluster.
```

File: modules/tutorials/pages/authentication_with_openldap.adoc [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+
+Use `stackablectl svc list` to find the endpoints of Superset and Trino and open their web interfaces in the Browser.
+
+You can log into Superset with user _admin_ and password _adminadmin_, and into Trino with user _admin_ and password _adminadmin_.
+
+These are the current users defined in Supersets and Trinos internal user management. Later you will see that these users cannot be used for authentication anymore after LDAP authentication has been enabled.
```
Comment (adwk67) [inline modules/tutorials/pages/authentication_with_openldap.adoc:35]:
```suggestion
These are the current users defined in Superset's and Trino's internal user management. Later you will see that these users cannot be used for authentication anymore after LDAP authentication has been enabled.
```

File: modules/tutorials/pages/authentication_with_openldap.adoc [hunk 7]
```diff
@@ -0,0 +1,334 @@
...
+// include::example$ldap-auth/30-install-openldap.sh[tag=apply-credentials-secretclass]
+
+This level of indirection is necessary, because the AuthenticationClass is cluster-scoped but Secrets are not.
+
+Now you can create the AuthenticationClass `openldap` which references the SecretClass:
+
+[source,yaml]
+include::example$ldap-auth/ldap-authenticationclass.yaml[]
+
+<1> The name of the AuthenticationClass, needs to be referenced later
```
Comment (adwk67) [inline modules/tutorials/pages/authentication_with_openldap.adoc:98]:
```suggestion
<1> The name of the AuthenticationClass, which needs to be referenced later
```

File: modules/tutorials/pages/authentication_with_openldap.adoc [hunk 8]
```diff
@@ -0,0 +1,334 @@
...
+    ...
+  workers:
+    ...
+  coordinators:
+    ...
+  ...
+----
+====
+
+You can now delete the TrinoCluster, we will recreate it later with the new configuration:
```
Comment (adwk67) [inline modules/tutorials/pages/authentication_with_openldap.adoc:226]:
```suggestion
You can now delete the TrinoCluster. We will recreate it later with the new configuration:
```

**Review by adwk67 — COMMENTED** [review]
Some minor grammatical things.

File: modules/tutorials/examples/ldap-auth/10-install-base.sh [hunk 9]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+echo "Installing the base stack"
+# tag::stackablectl-install[]
+stackablectl stack install trino-superset-s3
```
[the remark this answers is not in the document]
Comment (fhennig) [reply modules/tutorials/examples/ldap-auth/10-install-base.sh:6]: fixed [1f24545](https://github.com/stackabletech/documentation/pull/334/commits/1f24545e50dd4a354105d0106307d52e3529b834)

File: modules/tutorials/pages/authentication_with_openldap.adoc (hunk 4 above)
Comment (fhennig) [reply modules/tutorials/pages/authentication_with_openldap.adoc:13]: good point! I think it's important to be precise with language especially in tutorials

**Review by adwk67 — APPROVED** [review]
LGTM - nice tutorial: thanks!

**Pull request merged**